### PR TITLE
Frontend improvements

### PR DESCRIPTION
**Pull request by dglebovskiy**
- Added a main component to handle user auth state
- Moved note add/list navigation into a separate menu
- UserService now stores login state
- Registration modal now automatically logins the user after success
- Added a welcome page
- Unauthed users are redirected to a welcome page when attempting to change state

### CODE REVIEW

**Comment by evgeny-sav** [comment]
+1

File: notes_app/client/app/registration/registration.component.js [hunk 1]
```diff
@@ -7,7 +7,13 @@
     ctrl.attemptRegistration = function(username, password) {
       userService.register(username, password).then(function(status) {
         ctrl.registrationState['visible'] = false;
-        ctrl.close();
+        userService.login(username, password).then(function(status) {
+          ctrl.close({'$value': true});
+        }).catch(function(status) {
+          ctrl.registrationState['message'] = 'Problem logging in; try to log in manually'
```
Comment (ghost) [inline notes_app/client/app/registration/registration.component.js]: I'm not sure this is the bets message. `to log in manually` - what it supposes to mean? Maybe try to login later?

File: notes_app/client/app/login/login.component.js [hunk 2]
```diff
@@ -15,7 +15,7 @@
       });
     };
     ctrl.success = function() {
-      ctrl.close();
+      ctrl.close({'$value': true});
```
Comment (ghost) [inline notes_app/client/app/login/login.component.js]: could you please explain why you use `$value` here? it's not clear. Thanks

Comment (dglebovskiy) [reply notes_app/client/app/login/login.component.js]: It's a leftover from when the user state was stored in a main component. Now that the user state is stored in a service, it's no longer required and can be removed if necessary.

File: notes_app/client/app/registration/registration.component.js (hunk 1 above)
Comment (dglebovskiy) [reply notes_app/client/app/registration/registration.component.js]: Ok, will update.

**Review by ghost — APPROVED** [review]

**Pull request merged**